File: apps/merchant-api/InventoryService/src/InventoryService.Domain/Concretes/ProductVariant.cs
using InventoryService.Domain.Bases;

namespace InventoryService.Domain.Concretes;

public class ProductVariant : BaseEntity
{
    public Guid ProductId { get; set; }
    public Guid? ImageId { get; set; }
    public double PriceAdjustment { get; set; }
    public int StockQuantity { get; set; }
    public Product Product { get; set; }
    public Image Image { get; set; }
    public ICollection<ProductAttribute> Attributes { get; set; }
}

File: apps/merchant-web/src/commons/entities/concretes/ProductVariant.ts
import { UUID } from "crypto";
import Image from "./Image";
import ProductAttribute from "./ProductAttribute";

export default class ProductVariant {
  productVariantId: UUID;
  productId: UUID;
  productVariantImage: Image;
  priceAdjustment: number;
  stockQuantity: number;
  attributes: Array<ProductAttribute>;

  constructor(
    productVariantId: UUID,
    productId: UUID,
    productVariantImage: Image,
    priceAdjustment: number,
    stockQuantity: number,
    attributes: Array<ProductAttribute>
  ) {
    this.productVariantId = productVariantId;
    this.productId = productId;
    this.productVariantImage = productVariantImage;
    this.priceAdjustment = priceAdjustment;
    this.stockQuantity = stockQuantity;
    this.attributes = attributes;
  }
}
